File: apps/user/src/dao/RefreshSuggestionsDao.ts
'use strict'

import {SQS} from "aws-sdk";

const AWS = require("aws-sdk")

class RefreshSuggestionsDao {
    private sqs: SQS
    private readonly refreshSuggestionsQueueUrl: string

    constructor() {
        this.sqs = new AWS.SQS()
        // @ts-ignore
        this.refreshSuggestionsQueueUrl = process.env.REFRESH_SUGGESTIONS_QUEUE_URL
    }

    async sendRefreshSuggestionsRequest(userId: string) {
        const data = { userId }

        await this.sqs.sendMessage({
            QueueUrl: this.refreshSuggestionsQueueUrl,
            MessageBody: JSON.stringify(data)
        }).promise()
    }
}

export {RefreshSuggestionsDao}

File: apps/user/src/dao/refresh_suggestions_dao.py
import json
import os

import boto3
from types_boto3_sqs import SQSClient


class RefreshSuggestionsDao:
    def __init__(self):
        self.sqs: SQSClient = boto3.client("sqs")
        self.refresh_suggestions_queue = os.environ["REFRESH_SUGGESTIONS_QUEUE_URL"]

    def send_refresh_suggestions_request(self, user_id: str):
        self.sqs.send_message(
            QueueUrl=self.refresh_suggestions_queue,
            MessageBody=json.dumps({
                "userId": user_id,
            })
        )
